### PR TITLE
Fix variable list logging.

### LINKED ISSUE #521: Event logging doesn't handle the situation when the type of a variable ListType.
### What's your issue about?
Two programs:
```
Bar: __log__({_value: num[2]})

@public
def foo():
    log.Bar([1, 2])
```
```
Bar: __log__({_value: num[2]})

@public
def foo():
    a = [1, 2]
    log.Bar(a)
```
The first program compiles but the second program throws python exception "AttributeError: 'Name' object has no attribute 'elts'" because `pack_args_by_32` only handles list literals.

### How can it be fixed?

In `pack_args_by_32`, add logic to handle variable of ListType.

#### Cute Animal Picture

image

### PR DESCRIPTION
**Pull request by jacqueswww**
### - What I did

Fix #521, allowing logging using a variable of list type.

### - How I did it

Added section that copies list memory section into placeholder, see changes in `parser.py`.

### CODE REVIEW

File: tests/parser/features/test_logging.py [hunk 1]
```diff
@@ -448,3 +448,32 @@ def ioo(inp: bytes <= 100):
     c.ioo(b"moo4")
     assert s.head_state.receipts[-1].logs[0].data == b'moo4'
     print("Passed raw log tests")
+
+
```
Comment (DavidKnott) [inline tests/parser/features/test_logging.py]: I'd love to see a test with a list as function input.

Comment (DavidKnott) [inline tests/parser/features/test_logging.py]: Can you add a test with a list as function input?  Can you also add a test with a list of a different type?

**Comment by jacqueswww** [comment]
@DavidKnott thanks for the feedback. Also just a note, #524 might conflict with this one being in the same function, I will adapt this one on as necessary if that one merges to master.

**Comment by DavidKnott** [comment]
@jacqueswww Could you quick rebase this?

**Comment by DavidKnott** [comment]
LGTM, thanks a lot!

**Pull request merged**